src/platform/bm_lwip.c

```diff
@@ -121,16 +121,14 @@ static uint8_t bcmp_recv(void *arg, struct raw_pcb *pcb, struct pbuf *pbuf,
       // Make a copy of the IP address since we'll be modifying it later when we
       // remove the src/dest ports (and since it might not be in the pbuf so someone
       // else is managing that memory)
-      struct pbuf *p_ref = pbuf_alloc(PBUF_IP, pbuf->len, PBUF_RAM);
       ip_addr_t *src_ref = (ip_addr_t *)bm_malloc(sizeof(ip_addr_t));
       ip_addr_t *dst_ref = (ip_addr_t *)bm_malloc(sizeof(ip_addr_t));
       LwipLayout *layout = (LwipLayout *)bm_malloc(sizeof(LwipLayout));
-      pbuf_copy(p_ref, pbuf);
       memcpy(dst_ref, ip6_hdr->dest.addr, sizeof(ip_addr_t));
       memcpy(src_ref, src, sizeof(ip_addr_t));
-      *layout = (LwipLayout){p_ref, src_ref, dst_ref};
+      *layout = (LwipLayout){pbuf, src_ref, dst_ref};
 
-      BcmpQueueItem item = {BCMP_EVT_RX, (void *)layout, p_ref->len};
+      BcmpQueueItem item = {BCMP_EVT_RX, (void *)layout, layout->pbuf->len};
```

Comment:
I was getting a crash here part way through the clients download of the update where `p_ref` was NULL.